core/src/test/java/fr/sncf/osrd/envelope_sim/AllowanceRangesTests.java

```diff
@@ -237,4 +240,57 @@ public void testVeryShortRange() {
         assertEquals(marginTime2, targetTime2, 2 * TIME_STEP);
         assertEquals(marginTime3, targetTime3, 2 * TIME_STEP);
     }
+
+    /** Test with an allowance range that starts very slightly after the path start, and ends around the end
+     * of the acceleration part. This doesn't necessarily have to result in a valid envelope as we're very close
+     * to asking for an impossible allowance, but we check that it doesn't crash early */
+    @Test
+    public void testAllowanceRangeEdgeCase() {
+        var testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+        var length = 100_000;
+        var testPath = new FlatPath(length, 0);
+        var testContext = new EnvelopeSimContext(testRollingStock, testPath, TIME_STEP);
+        var stops = new double[] { testPath.getLength() };
+        var maxEffortEnvelope = makeSimpleMaxEffortEnvelope(testContext, 10, stops);
+        var value1 = new AllowanceValue.Percentage(10);
+        var rangesTransitions = new double[] { 0.1, maxEffortEnvelope.get(0).getEndPos() };
+        var ranges = List.of(
+                new AllowanceRange(rangesTransitions[0], rangesTransitions[1], value1)
+        );
+        var allowance = new LinearAllowance(
+                testContext, rangesTransitions[0], rangesTransitions[1], 0, ranges
+        );
+        try {
+            allowance.apply(maxEffortEnvelope);
+        } catch (AllowanceConvergenceException e) {
+            assertEquals(OSRDError.ErrorCause.USER, e.cause);
+        }
```

Comment:
The test passes even if no exception is thrown. I think you meant to use assertThrows

Reply:
That is what I meant, an "impossible range" error is a valid output here. No error is also valid.

When I wrote the test it caused an assert earlier, I didn't even know if the margin was possible